framework/lib/components/combo-picker/combo-picker.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from 'react'
+import { Button, InputGroup, InputRightElement, NumberInputField } from '@chakra-ui/react'
+import { identity } from 'ramda'
+import { ComboPickerProps } from './types'
+import { Select } from '../select'
+import { NumberInput } from '../number-input'
+
+/**
+ * @see ComboPicker
+ * @see {@link https://northlight.dev/reference/combo-picker}
+ */
+export const ComboPicker = ({
+  onChange = identity,
+  options,
+  defaultOption,
+  size,
+  value = { input: 0, option: defaultOption || options[0] },
+  placeholder,
+  ...rest
+}: ComboPickerProps) => {
+  const [ inputValue, setInputValue ] = useState(value.input)
+  const [ selectOption, setSelectOption ] = useState(value.option)
+  const [ isDropdownOpen, setIsDropdownOpen ] = useState(false)
+
+  const toggleDropdown = () => setIsDropdownOpen(!isDropdownOpen)
+  const hideDropdown = () => setIsDropdownOpen(false)
+
+  const handleInputChange = (newInputvalue: String) => {
+    setInputValue(Number(newInputvalue))
+  }
+
+  const handleSelectChange = (selectedOption: any) => {
+    if (selectedOption) {
+      setSelectOption(selectedOption)
+      hideDropdown()
+    }
+  }
+
+  useEffect(() => {
+    onChange({
+      input: inputValue, option: selectOption,
+    })
+  }, [ inputValue, selectOption ])
```

Comment:
any other way to do this. we try to avoid this pattern cause might cause alot of rerenders